src/Domain/FunctionsOfTime/ThreadsafeList.hpp

```diff
@@ -233,11 +235,35 @@ double ThreadsafeList<T>::expiration_after(const double time) const {
 
 template <typename T>
 void ThreadsafeList<T>::pup(PUP::er& p) {
+  size_t version = 0;
+  p | version;
+  // Remember to increment the version number when making changes to this
+  // function. Retain support for unpacking data written by previous versions
+  // whenever possible. See `Domain` docs for details.
+
+  if (version != 0) {
+    ERROR("Unrecognized version " << version);
+  }
+
   p | initial_time_;
-  p | interval_list_;
   if (p.isUnpacking()) {
+    bool empty{};
```

Comment:
Can you add a comment here that the reason we can't just do `p | interval_list_;` is because of thread-safety?